File: src/main/java/busanVibe/busan/domain/user/converter/UserConverter.java
package busanVibe.busan.domain.user.converter;

import busanVibe.busan.domain.user.domain.User;
import busanVibe.busan.domain.user.dto.TokenResponseDTO;
import busanVibe.busan.domain.user.dto.UserResponseDTO;
import org.springframework.stereotype.Component;

@Component
public class UserConverter {

  public UserResponseDTO.loginDto toLoginDto(
      User user, boolean isNewUser, TokenResponseDTO tokenResponseDTO) {

    return new UserResponseDTO.loginDto(
        user.getId(), tokenResponseDTO, user.getEmail(), isNewUser
    );
  }

}

File: src/main/kotlin/busanVibe/busan/domain/user/converter/UserConverter.kt
package busanVibe.busan.domain.user.converter

import busanVibe.busan.domain.user.data.User
import busanVibe.busan.domain.user.data.dto.login.TokenResponseDto
import busanVibe.busan.domain.user.data.dto.login.UserLoginResponseDTO
import org.springframework.stereotype.Component

@Component
class UserConverter {

    fun toLoginDto(
        user: User, isNewUser: Boolean, tokenResponseDto: TokenResponseDto
    ): UserLoginResponseDTO.LoginDto {
        return UserLoginResponseDTO.LoginDto(
            user.id, tokenResponseDto, user.email, isNewUser
        )
    }


}